src/controller/LoginController.ts

```diff
@@ -58,40 +60,27 @@ class LoginController {
     }
 
     /*
-     * The SPA posts its URL here on every page load, and this operation ends a login when required
-     * The API works out whether it is an OAuth response, eg:
-     * - code + state query parameters
-     * - code + error query parameters
-     * - JARM response parameters
+     * The SPA posts its URL here on every page load, to get its authenticated state
+     * When an OAuth response is received it is handled and cookies are written
      */
     handlePageLoad = async (req: express.Request, res: express.Response, next: express.NextFunction) => {
 
-        // Verify the web origin
         const options = new ValidateRequestOptions()
         options.requireCsrfHeader = false
         validateExpressRequest(req, options)
         
-        // First see if the SPA is reporting an OAuth front channel response to the browser
-        const data = this.getUrlParts(req.body?.pageUrl)
-        const isSuccessOAuthResponse = !!(data.state && data.code)
-        const isFailedOAuthResponse = !!(data.state && data.error)
-
-        // First handle reporting front channel errors back to the SPA
-        if (isFailedOAuthResponse) {
-
-            throw new AuthorizationResponseException(
-                data.error,
-                data.error_description || 'Login failed at the Authorization Server')
-        }
-
-        let isLoggedIn: boolean
+        const data = await handleAuthorizationResponse(req.body?.pageUrl)
+        
+        let isLoggedIn = false
+        let handled = false
         let csrfToken: string = ''
 
-        if (isSuccessOAuthResponse) {
+        if (data.code && data.state) {
             
-            // Main OAuth response handling
             const tempLoginData = req.cookies ? req.cookies[getTempLoginDataCookieName(config.cookieNamePrefix)] : undefined
+            
             const tokenResponse = await getTokenEndpointResponse(config, data.code, data.state, tempLoginData)
+            validateIDtoken(config, tokenResponse.id_token)
```

Comment:
The token response might not contain the ID token, so I think it should be validated only when present.